libs/components/core/src/lib/modules/content-descriptor-provider/content-descriptor-provider.ts

```diff
@@ -0,0 +1,20 @@
+import { Observable, ReplaySubject } from 'rxjs';
+
+/**
+ * @internal
+ * An API to provide a description of a parent component's content to child components.
+ * For example, toolbar can use this to provide its child components with a list
+ * descriptor they can use to construct aria labels, or tree view can provide the node
+ * name to its context menus.
+ */
+export class SkyContentDescriptorProvider {
```

Comment:
Paul mentioned use an argument object here. Do we want to make the name of this a little more generic and then have an interface object that we could add to later if needed?

Reply:
@Blackbaud-PaulCrowder what do you think? Because I changed the name to be ContentProvider instead of ContextProvider I'm having a hard time thinking of what other content-related info we might expand this to include, but it certainly wouldn't hurt

Reply:
Maybe `SkyContentInfoProvider` that has a `update()` method? That could take a `SkyContentInfo` parameter which has a property called `descriptor`.